### DISCUSSION TITLE
fluidnc and 6 pack connecting to ugs

### DISCUSSION THREAD

**Discussion by deanembrey** [winder/Universal-G-Code-Sender#2080] [Help]
ugs won't connect to 6 pack with fluidnc firmware. I select fluidnc and it gives an error. that the board is not supported. Are there any others having the same issue. It connects to grbl. thanks dean

**Comment by breiler** [comment]
What exactly does the error say, and how are you connecting?

The setup wizard does not yet support the firmware and will give you such a error.

If you connect to a controller that is running a really old version will also give you this error.

**Reply by deanembrey** [reply]
Been awhile have they updated to support for fluidnc. Latest version. What is the latest version of fluidnc ugs supports now?

**Reply by breiler** [reply]
I was testing out 3.6.6 this weekend and it seemed to function properly.

**Reply by deanembrey** [reply]
So I need to get the updated version ugs. What controller did you test it on? Thanks

**Reply by breiler** [reply]
I am running version UGS 2.0.13. The controller is based on on a ESP32 WROOM 32D. 
Which break out board doesn't really matter, but I have tested on both Barts `6-pack controller` and `ESP 32 CNC Dev Controller 4.1`

**Reply by deanembrey** [reply]
I wonder why it wasn't working on the 6 pack. Have you done an update on ugs since I last text you.

**Reply by breiler** [reply]
Sorry, I can not help you since you are not answering my questions...

Here is a list of alternatives to UGS:
http://wiki.fluidnc.com/en/support/gcode_senders

**Reply by deanembrey** [reply]
I got a screen shot of the error I get. It stops right there. But it does connect. If you know of some video that I can find to manually configure the stepper motor configuration. 
![image](https://user-images.githubusercontent.com/7685702/214246043-84fe4dbc-7c81-4edd-84a5-c942f11da767.png)

**Reply by deanembrey** [reply]
As you can see it does connect, and it will let me move the stepper motor. I guess I will have to live with that for now. thanks though.

**Comment by deanembrey** [comment]
I just bought the  6 pack. When I set it for fluidnc and try to connect it says its not compatible. I use to use with grbl32. Its will kind a connect though but is weird things happen. I am connecting with cable usb. I think bart put the newest fluidnc version. I just bought this a month ago. It has the esp32 processor plugged into the 6 pack controller from tindy store.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Reply by breiler** [reply]
Look, in order for us to be able to help you we need details. It is like if you are asking for help with a car and and you won't let us know if it is electric, gas or diesel...

If the board is running FluidNC the error message should contain the lowest expected version and the version your board firmware.  

> When I set it for fluidnc and try to connect it says its not compatible.

What exactly does it say?

>  Its will kind a connect though but is weird things happen. 

What do you mean by weird?

>  I think bart put the newest fluidnc version.

How does this help us figure out the problem?

**Reply by deanembrey** [reply]
Its a esp32 board. 6 pack from tindie by bart dring. When I try to connect I change the drop down menu to fluidnc. then i try to connect. and it gives me an error. It says not compatible. I have a window labtop, connected to a 6 pack esp32 controller. ugs tells me that my board is not compatible. I use fluidnc. Its the latest version. I think I have the latest version os ugs. It looks like it connects, though wont go past the error. the little icon comes together that show connection.

**Comment by deanembrey** [comment]
I thought ugs supported grbl firmware

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Reply by breiler** [reply]
It does, but you aren't running GRBL are you?

**Comment by deanembrey** [comment]
Fluidnc is grbl it’s the latest esp32 32 bit, they just renamed it. I am going to try a older java and see if that works.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Reply by breiler** [reply]
I am interested in which version of the FluidNC firmware you are running. For instance, the latest released version is v3.6.5. The oldest version that we support is 3.3.0.

The error message that says it is not compatible should contain your current version. That is why I am asking what the error message says...

Another Java version will not help you.

**Comment by deanembrey** [comment]
I think I got 3.51 or something, I need to hook it up again and run it and find out.I’ll check the error to. Sorry thanks for getting back so quick. The version was from sept. 19 2022. That’s when bart dring added the latest version. I say this because he was waiting for the next version to come out so he could install it. I’l let you know thanks dean
Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by deanembrey** [comment]
I am sorry I was at the dentist I will hook it back up again and right down errors word by word. I installed the new version and I run the wizard to connect and It only has the cancel button enabled. It seems like there is a flaw in the fluidnc upload that was done. That’s what I can tell you from here now. I ll read back the emails and answer all your questions find the anwer to them. Thanks for getting back to me. Dean embrey
nt from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Reply by breiler** [reply]
In my first response I wrote that the "Setup wizard" does not yet support FluidNC and you can not use that tool to set your machine up and calibrate it.

You should however be able to connect to the machine and use it:
https://github.com/winder/Universal-G-Code-Sender/wiki/Usage#connecting-to-the-machine

How to set your machine up and make calibrations you need to check the FluidNC manual:
http://wiki.fluidnc.com/
